### PR TITLE
[JIT] Add torch._C._jit submodule

### LINKED ISSUE #51691: jit internal python bindings all get added to the `torch.` namespace.
### The problem

We have a construct in the torch init that adds to `torch.XX` anything that is is `torch._C.XX`.
The jit init is given the torch._C module here: https://github.com/pytorch/pytorch/blob/45e5562fcc5b0759b26f46f18f1156213b354902/torch/csrc/jit/python/init.cpp#L134
This file adds a lot of internal stuff (_XX) which is ok as they are not added to `torch.`.

But the python IR binding called afterwards here: https://github.com/pytorch/pytorch/blob/45e5562fcc5b0759b26f46f18f1156213b354902/torch/csrc/jit/python/python_ir.cpp#L210

Adds many things with the `_` and so all of these things get added to `torch.` in the end.

Note that other subset of the jit might be doing the same thing, but I stumbled upon the IR binding one first.

### Proposed solution

My proposed solution is to make the top level module create a `torch._C._jit` submodule that will be passed to all the initialization code. This way, none of it will be added to the `torch.` package.
But we can explore other options as well!

cc @gmagogsfm

### PR DESCRIPTION
**Pull request by SplitInfinity**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#52910 [JIT] Add torch._C._jit submodule**

**Summary**
PR #52158 tried to move all JIT bindings from `torch._C` to a new
submodule `torch._C._jit`, but that...did not go well. This pull request
adds the new `torch._C._jit` submodule, but does not migrate the
existing bindings. Instead, it adds a unit test that fails if any new
bindings are added to `torch._C`. A comment in the test instructs
developers to add their new binding to the allowlist if it really should
be in `torch._C`, or to add it to the appropriate submodule (e.g
`torch._C._jit`, for example). The idea is to prevent the issue
described in #51691 from getting *worse* if it cannot be fixed.

**Test Plan**
Continuous integration.

**Fixes**
This commit fixes #51691.

Differential Revision: [D26698373](https://our.internmc.facebook.com/intern/diff/D26698373)

### CODE REVIEW

File: test/test_jit_bindings.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
+from test_jit import JitTestCase
+from torch.testing._internal.common_utils import run_tests
+
+import torch
+
+
+class TestBindings(JitTestCase):
```
Comment (SplitInfinity) [inline test/test_jit_bindings.py:7]: Will this test actually get run? Do I need to add it to some other file or lists of tests?

Comment (albanD) [reply test/test_jit_bindings.py:7]: You need to add it to this list: https://github.com/pytorch/pytorch/blob/7a178a8a523d4653a3a2fa10c573b71e7fab1b9a/test/run_test.py#L20

**Review by albanD — APPROVED** [review]
This looks great! Thanks a lot for doing this.

I think this is not actually specific to jit right?
Maybe we can call this `test_public_cpp_bindings.py` ?

But even if you don't have time to do the rename/update the doc, this is completely worth having!

**Review by albanD — APPROVED** [review]
Thanks for the update. Looks good to go!

**Pull request closed without merging**